### PR TITLE
WT-4281 Reduce Python test suite runtime

### PR DESCRIPTION
**Pull request by agorrod**
By reducing runtime for longest running tests.

### CODE REVIEW

**Comment by agorrod** [comment]
autobuild: retest this please

**Comment by agorrod** [comment]
autobuild: retest this please

**Comment by agorrod** [comment]
autobuild: retest this please

**Comment by nehakhatri5** [comment]
autobuild: retest this please

**Comment by lukech** [comment]
retest this please

File: test/suite/test_backup02.py [hunk 1]
```diff
@@ -29,15 +29,17 @@
...
 class test_backup02(wttest.WiredTigerTestCase):
-    scenarios = make_scenarios([
-        ('table', dict(uri='table:test',fmt='L',dsize=100,nops=200,nthreads=1,time=30)),
-    ])
+    uri = 'table:test_backup02'
+    fmt = 'L'
+    dsize = 100
+    nops = 200
+    nthreads = 1
+    time = 60 if wttest.islongtest() else 10
```
Comment (sueloverso) [inline test/suite/test_backup02.py]: Do you think that doubling the time in the long test buys us more coverage or better testing? I'd be okay if the long test used `30`.

File: test/suite/test_bug010.py [hunk 2]
```diff
@@ -37,7 +37,7 @@
 class test_bug010(wttest.WiredTigerTestCase):
     name = 'test_bug010'
     uri = 'table:' + name
-    num_tables = 1000
+    num_tables = 2000 if wttest.islongtest() else 100
```
Comment (sueloverso) [inline test/suite/test_bug010.py]: (Again, I'm not sure if doubling the original for the long test adds better testing.)
Running this as a short test is not meaningful. I ran your modified test and the current test against the original fix (changeset `4235c69`) and its unfixed parent (changeset `8ab5d1ce`). (Had to fix up the `set_item` changes for cursor assignment.)

The current test (from develop) fails in the parent and passes in the fixed branch. Your modified/shortened test passes in both. So your short version is no longer testing the bug.

So leave as-is or maybe make the test long-only, but this form is not helpful.

File: test/suite/test_bug019.py [hunk 3]
```diff
@@ -35,15 +35,17 @@
...
 
     # Modify rows so we write log records. We're writing a lot more than a
     # single log file, so we know the underlying library will churn through
     # log files.
     def populate(self, nentries):
         c = self.session.open_cursor(self.uri, None, None)
         for i in range(0, nentries):
-            c[i] = i
+            # Make the values about 200 bytes - that's about 1MB of data for
+            # 5000 records, so at 10 log files used plus more for overhead.
```
Comment (sueloverso) [inline test/suite/test_bug019.py]: I suggest 'so at' -> 'generating'.
I also suggest 'about 200 bytes - that's...' -> 'about 200 bytes. That's...'

File: test/suite/test_config07.py [hunk 4]
```diff
@@ -60,7 +60,9 @@ class test_config07(wttest.WiredTigerTestCase):
     def populate(self):
         cur = self.session.open_cursor(self.uri, None, None)
         for i in range(0, self.entries):
-            cur[i] = i
+            # Make the values about 200 bytes - that about 1MB of data for
```
Comment (sueloverso) [inline test/suite/test_config07.py]: Same as above. '200 bytes. That's...'

File: test/suite/test_sweep03.py [hunk 5]
```diff
@@ -56,6 +56,24 @@ class test_sweep03(wttest.WiredTigerTestCase, suite_subprocess):
 
     scenarios = make_scenarios(types)
 
+    # Wait for the sweep server to run - let it run twice, since the statistic
+    # is incrememented at the start of a sweep and the test relies on sweep
+    # completing it's work
```
Comment (sueloverso) [inline test/suite/test_sweep03.py]: End with a period.

File: test/suite/test_sweep03.py [hunk 6]
```diff
@@ -56,6 +56,24 @@ class test_sweep03(wttest.WiredTigerTestCase, suite_subprocess):
...
+        for i in range(10):
+            stat_cursor = self.session.open_cursor('statistics:', None, None)
+            sweeps = stat_cursor[stat.conn.dh_sweeps][2]
+            stat_cursor.close()
+            if (sweeps > baseline + 1):
+                return sweeps
+            time.sleep(0.5)
+
+        # If the statistic didn't increase in 5 seconds the sweep server isn't
+        # working as expected
```
Comment (sueloverso) [inline test/suite/test_sweep03.py]: End sentence with a period.

File: test/suite/test_sweep03.py [hunk 7]
```diff
@@ -72,18 +90,14 @@ def test_disable_idle_timeout1(self):
 
         #
         # The idle timeout is disabled - we don't expect the sweep server to
-        # close any regular handles. Give the sweep server 5x the scan
-        # interval to close any handles.
+        # close any regular handles.
         #
-        time.sleep(5)
+        self.wait_for_sweep(0)
```
Comment (sueloverso) [inline test/suite/test_sweep03.py]: The function returns a value but you don't assign anything here. Maybe assign it to `unused` and add a comment.

Comment (agorrod) [reply test/suite/test_sweep03.py]: I've updated the comment - I don't think always assigning returns is idiomatic in Python - so I didn't add an unused variable.

File: test/suite/test_bug010.py (hunk 2 above)
Comment (agorrod) [reply test/suite/test_bug010.py]: I did a similar exercise, and saw that it fails reliably between 120 and 140 for me - so I've moved the lower number to 200.

**Comment by agorrod** [comment]
autobuild: retest this please

**Comment by agorrod** [comment]
I'm going to go ahead and merge this one - it's had a few rounds of review, and we can do follow on work to further reduce runtime if we get time.

**Pull request merged**